File: front_end/src/app/Services/beneficiaire.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { Adherant } from '../Models/Adherant';
import { Assurance } from '../Models/Assurance';
import { Beneficiaire } from '../Models/Beneficiaire';

@Injectable({
  providedIn: 'root'
})

export class BeneficiaireService {

  private baseUrl = 'http://localhost:8034/api/beneficiaires' ;

  constructor(private http: HttpClient) { }

  getAllBeneficiaires(): Observable<Beneficiaire[]> {
    return this.http.get<Beneficiaire[]>(this.baseUrl);
  }

  getBeneficiaireById(id: number): Observable<Beneficiaire> {
    return this.http.get<Beneficiaire>(`${this.baseUrl}/${id}`);
  }

  createBeneficiaire(adherantId: number , beneficiaire: Beneficiaire): Observable<Beneficiaire> {
    return this.http.post<Beneficiaire>(`${this.baseUrl}/adherant/${adherantId}`, beneficiaire);
  }

  updateBeneficiaire(id: number, beneficiaire: Beneficiaire, idBenef: number): Observable<Beneficiaire> {
    return this.http.put<Beneficiaire>(`${this.baseUrl}/update/${id}/${idBenef}`, beneficiaire);
  }


  deleteBeneficiaire(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

}

File: backend/src/main/java/com/example/projectPfe/services/Interface/BeneficiaireService.java
package com.example.projectPfe.Services.Interface;

import com.example.projectPfe.models.Adherant;
import com.example.projectPfe.models.Beneficiaire;

import java.util.List;
import java.util.Optional;

public interface BeneficiaireService {

    List<Beneficiaire> getAllBeneficiaires();
    Optional<Beneficiaire> getBeneficiaireById(int id);
    Beneficiaire saveOrUpdateBeneficiaire(Beneficiaire beneficiaire);
    List<Beneficiaire> getBeneficiaireByAdherant(Adherant adherant);
    void deleteBeneficiaire(int id);
    Beneficiaire addBeneficiaireToAdherant(Beneficiaire beneficiaire, Adherant adherant);
}
